src/types.jl

```diff
@@ -9,7 +9,30 @@ abstract AbstractOID
 type OID{N} <: AbstractOID end
 
 oid{T<:AbstractPostgresType}(t::Type{T}) = convert(OID, t)
-pgtype(t::Type) = convert(PostgresType, t)
+
+if VERSION < v"0.5-dev+4194"
+    import Compat
+
+    function pgtype(t::Type)
```

Comment:
Rather than an if, these can just be three methods of `pgtype` (e.g. `pgtype{T<:Compat.ASCIIString}(::Type{T})`